datafusion/sql/src/expr/function.rs

```diff
@@ -392,10 +392,6 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
         } else {
             // User defined aggregate functions (UDAF) have precedence in case it has the same name as a scalar built-in function
             if let Some(fm) = self.context_provider.get_aggregate_meta(&name) {
-                if fm.is_ordered_set_aggregate() && within_group.is_empty() {
```

Comment:
there appears to be no tests for this error nor any reason I can see for not allowing it